ambry-clustermap/src/main/java/com.github.ambry.clustermap/AmbryPartition.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.clustermap;
+
+import com.github.ambry.utils.Utils;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.util.List;
+
+
+/**
+ * {@link PartitionId} implementation to use within dynamic cluster managers.
+ */
+class AmbryPartition extends PartitionId {
+  private final Long id;
+  private final HelixClusterManager.HelixClusterManagerCallback clusterManagerCallback;
+  private volatile PartitionState state;
+
+  private static final short VERSION_FIELD_SIZE_IN_BYTES = 2;
+  private static final short CURRENT_VERSION = 1;
+  private static final int PARTITION_SIZE_IN_BYTES = VERSION_FIELD_SIZE_IN_BYTES + 8;
+
+  /**
+   * Instantiate an AmbryPartition instance.
+   * @param id the id associated with this partition.
+   * @param clusterManagerCallback the {@link HelixClusterManager.HelixClusterManagerCallback} to use to make callbacks
+   *                               to the {@link HelixClusterManager}
+   * The initial state defaults to {@link PartitionState#READ_WRITE}.
+   */
+  AmbryPartition(long id, HelixClusterManager.HelixClusterManagerCallback clusterManagerCallback) {
+    this.id = id;
+    this.clusterManagerCallback = clusterManagerCallback;
+    this.state = PartitionState.READ_WRITE;
+  }
+
+  @Override
+  public byte[] getBytes() {
+    ByteBuffer buffer = ByteBuffer.allocate(PARTITION_SIZE_IN_BYTES);
```

Comment:
Moved this to a common method in `ClusterMapUtils`. This is just serializing a short and a long.